src/Symfony/Component/Security/Core/User/UserInterface.php

```diff
@@ -47,7 +46,7 @@
      * and populated in any number of different ways when the user object
      * is created.
      *
-     * @return (Role|string)[] The user roles
+     * @return Role|string[] The user roles
```

Comment:
It should stay `(Role|string)[]` it's an array that can contains a mix of both Role objects and strings